### PR TITLE
Experiments results recording

### PR DESCRIPTION
**Pull request by dhruvtapasvi**

### CODE REVIEW

File: src/evaluation/metric/BinaryCrossEntropy.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+from evaluation.metric.Metric import Metric
+from evaluation.metric.MetricResult import MetricResult
+import numpy as np
+import math
+# from keras.metrics import binary_crossentropy
```
Comment (dhruvtapasvi) [inline src/evaluation/metric/BinaryCrossEntropy.py]: Remove this comment

File: src/evaluation/metric/BinaryCrossEntropy.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+from keras.backend import variable, binary_crossentropy, sum, eval
+
+
+
```
Comment (dhruvtapasvi) [inline src/evaluation/metric/BinaryCrossEntropy.py]: Too many blank lines

File: src/evaluation/metric/BinaryCrossEntropy.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+class BinaryCrossEntropy(Metric):
+    def compute(self, first: np.ndarray, second: np.ndarray) -> MetricResult:
+        firstFlattened = variable(first.reshape((first.shape[0], -1)))
```
Comment (dhruvtapasvi) [inline src/evaluation/metric/BinaryCrossEntropy.py:9]: len(first), len(second)

File: src/evaluation/metric/MetricResult.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import numpy as np
+
+
+class MetricResult:
+    def __init__(self, allValues: np.ndarray):
+        self.__allValues = allValues
+        self.__mean = np.mean(allValues)
+        self.__standardDeviation = np.std(allValues)
```
Comment (dhruvtapasvi) [inline src/evaluation/metric/MetricResult.py]: Sample std I think should be over here

File: src/evaluation/metric/SquaredError.py [hunk 5]
```diff
@@ -7,7 +8,7 @@ class SquaredError(Metric):
     def __init__(self, variance):
         self.__variance = variance
 
-    def compute(self, first: np.ndarray, second: np.ndarray):
+    def compute(self, first: np.ndarray, second: np.ndarray) -> MetricResult:
         firstFlattened = first.reshape((first.shape[0], -1))
```
Comment (dhruvtapasvi) [inline src/evaluation/metric/SquaredError.py:12]: len(first)

File: src/experiment/InterpolateExperiment.py [hunk 6]
```diff
@@ -1,51 +1,99 @@
...
+                self.__imageSpaceComparisonMetric.compute(interpolateSubdataset.xCentre, interpolatedReconstructed).mean
+            )
+
+            self.__resultsStore.storeValue(
+                [self.__datasetName, self.__modelName, interpolateSubdataset.interpolatedFactorName, "actual", "metricLatentSpace"],
+                self.__latentSpaceComparisonMetric.compute(
+                    self.__autoencoder.encoder().predict(interpolateSubdataset.xCentre, batch_size=100),
+                    interpolated
+                ).mean
+            )
```
Comment (dhruvtapasvi) [inline src/experiment/InterpolateExperiment.py]: Add interpolation in the image space as a point of comparison, and also random picture

File: src/experiment/InterpolateExperiment.py [hunk 7]
```diff
@@ -1,51 +1,99 @@
...
+    def __numericalInterpolationMetrics(self, interpolateSubdataset: InterpolateSubdataset):
+        interpolated, interpolatedReconstructed = self.__interpolateLatentSpace.interpolateAll(interpolateSubdataset.xLeft, interpolateSubdataset.xRight, 2)
+
+        interpolatedReconstructed = interpolatedReconstructed[:, 1]
+        interpolated = interpolated[:, 1]
+
+        if interpolateSubdataset.centreIsSpecified():
+            self.__resultsStore.storeValue(
+                [self.__datasetName, self.__modelName, interpolateSubdataset.interpolatedFactorName, "actual", "metricImageSpace"],
+                self.__imageSpaceComparisonMetric.compute(interpolateSubdataset.xCentre, interpolatedReconstructed).mean
```
Comment (dhruvtapasvi) [inline src/experiment/InterpolateExperiment.py]: Don't only store the mean
